src/services/Updates.php

```diff
@@ -366,7 +367,7 @@ public function updateCraftVersionInfo(): bool
         // Only update the schema version if it's changed from what's in the file,
         // so we don't accidentally overwrite other pending changes
         $projectConfig = Craft::$app->getProjectConfig();
-        if ($projectConfig->get(ProjectConfig::PATH_SCHEMA_VERSION, true) !== $info->schemaVersion) {
```

Comment:
This prevents `system.schemaVersion` from getting updated when you’re updating for the first time (e.g. from a dev environment), since the value will match what’s in the `info` table at that point.

Reply:
Yes, you’re right. I must have got confused when testing. I do wonder if that check is needed, though. The `ProjectConfig->_setInternal()` method will check if the value has changed anyway, so dropping that check would mean that the value would only get updated when needed anyway (first update and when deploying further).

Those checks were added [a while ago](https://github.com/craftcms/cms/commit/68b05761b439b0bc42bf4ebc3572ceb8cbdfc2ae), and it looks like back then we weren’t checking if the value has changed before setting it.

Reply:
Yeah, maybe not needed as of af448b70336.

Still, not sure I’m following how #16086 could occur because of these checks. Were you able to reliably reproduce it? If so can you update the PR with steps to reproduce?

Reply:
Sure! I update the description with the steps to reproduce. 

I also pushed the change to let the pc `_setInternal()` figure out if the schema version value has changed.